src/EPR.Calculator.API/Mappers/CreateDefaultParameterSettingMapper.cs

```diff
@@ -1,6 +1,36 @@
-﻿namespace api.Mappers
+﻿using EPR.Calculator.API.Data.DataModels;
+using EPR.Calculator.API.Dtos;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace api.Mappers
 {
     public static class CreateDefaultParameterSettingMapper
     {
+        public static List<DefaultSchemeParametersDto> Map(DefaultParameterSettingMaster defaultParameterSettingMaster, DbSet<DefaultParameterTemplateMaster> defaultParameterTemplate)
+        {
+            var result = new List<DefaultSchemeParametersDto>();
+
+            foreach (var item in defaultParameterSettingMaster.Details)
+            {
+                var data = new DefaultSchemeParametersDto
+                {
+                    Id = item.Id,
+                    ParameterYear = defaultParameterSettingMaster.ParameterYear,
+                    EffectiveFrom = defaultParameterSettingMaster.EffectiveFrom,
+                    EffectiveTo = defaultParameterSettingMaster.EffectiveTo,
+                    CreatedBy = defaultParameterSettingMaster.CreatedBy,
+                    CreatedAt = defaultParameterSettingMaster.CreatedAt,
+                    DefaultParameterSettingMasterId = item.Id,
+                    ParameterUniqueRef = item.ParameterUniqueReferenceId,
+                    ParameterType = defaultParameterTemplate.FirstOrDefault(x => x.ParameterUniqueReferenceId == item.ParameterUniqueReferenceId)?.ParameterType,
+                    ParameterCategory = defaultParameterTemplate.FirstOrDefault(x => x.ParameterUniqueReferenceId == item.ParameterUniqueReferenceId)?.ParameterCategory,
```

Comment:
Should this be defaultParameterTemplate.Single?
And also if you are using it multiple time, You could create a variable for it

Reply:
I have updated that Uday